CMakeLists.txt

```diff
@@ -451,14 +444,39 @@ endif(BUILD_DOC)
 # Python
 # ----------------------------------------------------------------------------
 option(ENABLE_PYTHON "Create python library via SWIG" ON)
-find_package(PythonInterp)
-find_package(PythonLibs)
 
-if(ENABLE_PYTHON AND PYTHONLIBS_FOUND)
+find_package(Python 3.0
+  REQUIRED COMPONENTS Interpreter Development
+  OPTIONAL_COMPONENTS NumPy)
```

Comment:
Here numpy is only an optional component. 

I was not able to complie CRPropa with Python but without numpy. 
And on the other hand we state on our installation documentation, that numpy is required. Therefore I would suggest to set it to the REQUIRED COMPONENTS. 

If we are strictly with the numpy dependence a lot of code checking if numpy is available or not can be skiped, and it gets better readable